build.zig

```diff
@@ -0,0 +1,1484 @@
+    try orca_platform_files.append("src/orca.c");
+
+    orca_platform_lib.addLibraryPath(b.path("build/bin"));
+
+    if (target.result.os.tag == .windows) {
+        orca_platform_lib.linkSystemLibrary("user32");
+        orca_platform_lib.linkSystemLibrary("opengl32");
+        orca_platform_lib.linkSystemLibrary("gdi32");
+        orca_platform_lib.linkSystemLibrary("dxgi");
+        orca_platform_lib.linkSystemLibrary("dxguid");
+        orca_platform_lib.linkSystemLibrary("d3d11");
+        orca_platform_lib.linkSystemLibrary("dcomp");
+        orca_platform_lib.linkSystemLibrary("shcore");
+        // orca_platform_lib.linkSystemLibrary("delayimp");
+        orca_platform_lib.linkSystemLibrary("dwmapi");
+        orca_platform_lib.linkSystemLibrary("comctl32");
+        orca_platform_lib.linkSystemLibrary("ole32");
+        orca_platform_lib.linkSystemLibrary("shell32");
+        orca_platform_lib.linkSystemLibrary("shlwapi");
+
+        orca_platform_lib.linkSystemLibrary("libEGL.dll"); // todo DELAYLOAD?
+        orca_platform_lib.linkSystemLibrary("libGLESv2.dll"); // todo DELAYLOAD?
+        orca_platform_lib.linkSystemLibrary("webgpu"); // todo DELAYLOAD?
+    } else if (target.result.os.tag.isDarwin()) {
+        orca_platform_lib.linkFramework("Carbon");
+        orca_platform_lib.linkFramework("Cocoa");
+        orca_platform_lib.linkFramework("Metal");
+        orca_platform_lib.linkFramework("QuartzCore");
+        orca_platform_lib.linkFramework("UniformTypeIdentifiers");
+
+        // TODO add @rpath stuff?
+        // orca_platform_lib.addRPath("@rpath");
+        orca_platform_lib.linkSystemLibrary2("EGL", .{ .weak = true });
+        orca_platform_lib.linkSystemLibrary2("GLESv2", .{ .weak = true });
+        orca_platform_lib.linkSystemLibrary2("webgpu", .{ .weak = true });
+
+        try orca_platform_files.append("src/orca.m");
+    }
+
+    orca_platform_lib.addCSourceFiles(.{
+        .files = orca_platform_files.items,
+        .flags = orca_platform_compile_flags.items,
+    });
+
+    orca_platform_lib.step.dependOn(&stage_angle_artifacts.step);
+    orca_platform_lib.step.dependOn(&stage_dawn_artifacts.step);
+    orca_platform_lib.step.dependOn(&update_wgpu_header.step);
+
+    orca_platform_lib.step.dependOn(&orca_runtime_bindgen_core.step);
+    orca_platform_lib.step.dependOn(&orca_runtime_bindgen_surface.step);
+    orca_platform_lib.step.dependOn(&orca_runtime_bindgen_clock.step);
+    orca_platform_lib.step.dependOn(&orca_runtime_bindgen_io.step);
+    orca_platform_lib.step.dependOn(&orca_runtime_bindgen_gles.step);
+
+    const orca_platform_install: *Build.Step.InstallArtifact = b.addInstallArtifact(orca_platform_lib, .{});
+
+    const build_orca_platform_step = b.step("orca-platform", "Build the Orca platform layer from source.");
+    build_orca_platform_step.dependOn(&orca_platform_install.step);
+
+    // wasm3
+
+    var wasm3_sources = CSources.init(b);
```

Comment:
Consider explicitly enumerating all C file build inputs as opposed to automatically scraping them

Reply:
done